File: src/main/java/frc/team4069/robot/subsystems/WinchSubsystem.java
package frc.team4069.robot.subsystems;

import com.ctre.phoenix.motorcontrol.ControlMode;
import frc.team4069.robot.motors.TalonSRXMotor;

public class WinchSubsystem extends SubsystemBase {
    private static WinchSubsystem instance;

    private TalonSRXMotor talon;

    private WinchSubsystem() {
        talon = new TalonSRXMotor(10, 17);
    }

    public void start(boolean reversed) {
        talon.set(ControlMode.PercentOutput, reversed ? -1 : 1);
    }

    public void stop() {
        talon.stop();
    }

    public static WinchSubsystem getInstance() {
        if(instance == null) {
            instance = new WinchSubsystem();
        }

        return instance;
    }
}

File: src/main/java/frc/team4069/robot/subsystems/WinchSubsystem.kt
package frc.team4069.robot.subsystems

import frc.team4069.robot.motors.TalonMotor

object WinchSubsystem : SubsystemBase() {
    private val talon = TalonMotor(10, slaves = *intArrayOf(17))

    fun reset() {
        talon.stop()
    }

    fun start(reversed: Boolean = false) = talon.setConstantSpeed(if(reversed) -0.75 else 0.75)
    fun stop() = talon.stop()
}